src/migrate_411_412.ml

```diff
@@ -0,0 +1,1203 @@
+  Ast_411.Asttypes.mutable_flag -> Ast_412.Asttypes.mutable_flag =
+  function
+  | Ast_411.Asttypes.Immutable -> Ast_412.Asttypes.Immutable
+  | Ast_411.Asttypes.Mutable -> Ast_412.Asttypes.Mutable
+and copy_variance : Ast_411.Asttypes.variance -> Ast_412.Asttypes.variance =
+  function
+  | Ast_411.Asttypes.Covariant -> Ast_412.Asttypes.Covariant
+  | Ast_411.Asttypes.Contravariant -> Ast_412.Asttypes.Contravariant
+  | Ast_411.Asttypes.Invariant -> Ast_412.Asttypes.NoVariance
+and copy_value_description :
+  Ast_411.Parsetree.value_description -> Ast_412.Parsetree.value_description
+  =
+  fun
+    { Ast_411.Parsetree.pval_name = pval_name;
+      Ast_411.Parsetree.pval_type = pval_type;
+      Ast_411.Parsetree.pval_prim = pval_prim;
+      Ast_411.Parsetree.pval_attributes = pval_attributes;
+      Ast_411.Parsetree.pval_loc = pval_loc }
+    ->
+    {
+      Ast_412.Parsetree.pval_name = (copy_loc (fun x -> x) pval_name);
+      Ast_412.Parsetree.pval_type = (copy_core_type pval_type);
+      Ast_412.Parsetree.pval_prim = (List.map (fun x -> x) pval_prim);
+      Ast_412.Parsetree.pval_attributes = (copy_attributes pval_attributes);
+      Ast_412.Parsetree.pval_loc = (copy_location pval_loc)
+    }
+and copy_object_field_desc :
+  Ast_411.Parsetree.object_field_desc -> Ast_412.Parsetree.object_field_desc
+  =
+  function
+  | Ast_411.Parsetree.Otag (x0, x1) ->
+      Ast_412.Parsetree.Otag ((copy_loc copy_label x0), (copy_core_type x1))
+  | Ast_411.Parsetree.Oinherit x0 ->
+      Ast_412.Parsetree.Oinherit (copy_core_type x0)
+and copy_arg_label : Ast_411.Asttypes.arg_label -> Ast_412.Asttypes.arg_label
+  =
+  function
+  | Ast_411.Asttypes.Nolabel -> Ast_412.Asttypes.Nolabel
+  | Ast_411.Asttypes.Labelled x0 -> Ast_412.Asttypes.Labelled x0
+  | Ast_411.Asttypes.Optional x0 -> Ast_412.Asttypes.Optional x0
+and copy_closed_flag :
+  Ast_411.Asttypes.closed_flag -> Ast_412.Asttypes.closed_flag =
+  function
+  | Ast_411.Asttypes.Closed -> Ast_412.Asttypes.Closed
+  | Ast_411.Asttypes.Open -> Ast_412.Asttypes.Open
+and copy_label : Ast_411.Asttypes.label -> Ast_412.Asttypes.label =
+  fun x -> x
+and copy_rec_flag : Ast_411.Asttypes.rec_flag -> Ast_412.Asttypes.rec_flag =
+  function
+  | Ast_411.Asttypes.Nonrecursive -> Ast_412.Asttypes.Nonrecursive
+  | Ast_411.Asttypes.Recursive -> Ast_412.Asttypes.Recursive
+and copy_constant : Ast_411.Parsetree.constant -> Ast_412.Parsetree.constant
+  =
+  function
+  | Ast_411.Parsetree.Pconst_integer (x0, x1) ->
+      Ast_412.Parsetree.Pconst_integer (x0, (Option.map (fun x -> x) x1))
+  | Ast_411.Parsetree.Pconst_char x0 -> Ast_412.Parsetree.Pconst_char x0
+  | Ast_411.Parsetree.Pconst_string (x0, x1, x2) ->
+      Ast_412.Parsetree.Pconst_string
+        (x0, (copy_location x1), (Option.map (fun x -> x) x2))
+  | Ast_411.Parsetree.Pconst_float (x0, x1) ->
+      Ast_412.Parsetree.Pconst_float (x0, (Option.map (fun x -> x) x1))
+and copy_Longident_t : Longident.t -> Longident.t =
```

Comment:
In previous migrations, this was the identity. Do you know why it is not anymore?

Reply:
Same for `Location.t` and `Lexing.position`.

Reply:
Those conversions are auto-generated. Maybe someone made a manual pass in previous versions to clean up performance by making them identities. I can do that here.

Reply:
That's quite possible. Yeah, let's do the same here to be on the conservative side. Rest of the PR looks good

Reply:
Okay, I restored these conversions, plus `copy_loc_stack`, to be the identity.